src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,106 @@
  * limitations under the License.
  */
 
+import {execSync} from 'child_process';
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
 
 class ToolchainCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getToolchains(_toolchainType: string, _start?: number, _count?: number): Toolchains {
+    if (parseInt(execSync(`apt-cache show ${this.toolchainName} > /dev/null 2>&1; echo $?`)
+                     .toString()
+                     .trim()) !== 0) {
+      throw Error('the prerequisites are not met');
+    }
+
+    const availableToolchains = new Toolchains();
+
+    if (_toolchainType === 'latest') {
+      const versionList =
+          execSync(`apt-cache show ${this.toolchainName} | grep Version | awk '{print $2}'`)
+              .toString()
+              .trim()
+              .split('\n');
+
+      for (const version of versionList) {
+        const [major, minor, patch, option] = version.split(/[.~]+/);
```

Comment:
Please check whether this split function is satisfied with all the following versions.

```
Given version => Parsing version ([major,minor,patch,option])
1 => 1, 0, 0, ''
1.2 => 1, 2, 0, ''
1.2.3 => 1, 2, 3, ''
1.0~RC0 => 1, 0, 0, '~RC0'
1.0-beta => 1, 0, 0, '-beta'
1.1.0~RC0 => 1, 1, 0, '~RC0'
1.2.0~123456 => 1, 2, 0, '~123456'
```

Reply:
Should we write about this in test code?

Reply:
If you implement it as a function separately, you can add some tests about this. :)

Reply:
I've seen some other debian versions which includes `+` or `:`  sign. Should we care about them in `parseVersion`? or could they be some negative cases for our codes?